File: python/erat.py
import sys
max = int(sys.argv[1])
from datetime import datetime
t = datetime.now()

sieve = []
for i in range(max):
    sieve.append(True);
sieve[0] = False;
sieve[1] = False;
from math import sqrt
lim = int(sqrt(max))
for n in range(2, lim) :
    if sieve[n] :
        for j in range(2*n, max) :
            sieve[j] = False
            j += n

t = datetime.now() - t
print 'time: ' + str(t)

File: node/erat.js
/* Sieve of Eratosthenes program in Node.js */
console.log("erat");

var max = parseInt(process.argv[2])
console.log("max: " + max)

var t1 = Date.now()

var sieve = new Array(max)
//console.log(sieve.length)
sieve[0] = sieve[1] = false

for(var i = 2; i < max; i++) {
  sieve[i] = true
}

var lim = Math.trunc(Math.sqrt(max))
//console.log(lim)

for (n = 2; n < lim; n++) {
  if (sieve[n]) {
    for (j = 2 * n; j < max; j += n) {
      sieve[j] = false
    }
  }
}

//console.log(sieve)
var t2 = Date.now()
var tt = t2 - t1
console.log(tt + " ms")
